src/SqlQuery.php

```diff
@@ -25,17 +25,27 @@
 use function is_array;
 use function is_object;
 use function json_encode;
-use function preg_replace;
+use function preg_match;
 use function sprintf;
-use function stripos;
 use function strpos;
 use function trim;
 
 use const JSON_THROW_ON_ERROR;
 
 final class SqlQuery implements SqlQueryInterface
 {
-    private const C_STYLE_COMMENT = '/\/\*(.*?)\*\//u';
+    // Pattern to detect SELECT queries by skipping comments and finding the first SQL keyword
+    // ^ : Start from beginning of string
+    // \s* : Skip leading whitespace
+    // (?:...)* : Non-capturing group repeated 0 or more times (comment blocks)
+    //   \/\*.*?\*\/ : C-style comment /* ... */
+    //   | : OR operator
+    //   --.*?[\r\n] : Hyphen comment -- to end of line
+    //   \s* : Skip whitespace after comments
+    // \s* : Skip final whitespace
+    // (SELECT|WITH) : Capture SELECT or WITH keywords (read-only queries)
+    // i : Case insensitive
+    private const SELECT_QUERY_PATTERN = '/^\s*(?:\/\*.*?\*\/\s*|--.*?(?:[\r\n]|$)\s*)*\s*(SELECT|WITH)/i';
```

Comment:
```suggestion
    private const SELECT_QUERY_PATTERN = '/^\s*(?:(?:--[^\r\n]*|\/\*.*?\*\/|\(+)\s*)*(SELECT|WITH)\b/is';
```
正規表現こちらでどうでしょうか？
ややイレギュラーですが念のため、 `(SELECT ...) UNION (SELECT ...)` に対応しました。
変更点以下にまとめました。

| 項目 | 元の正規表現 | 変更版 | 問題点（旧） | 改善点（新）| 
| -- | -- | -- | -- | --| 
| 1. 括弧への対応 | 括弧を無視 | \(+ で括弧を許容 | (SELECT ...) にマッチしない | 括弧の数やネストに関係なく許容されるようにした| 
| 2. コメントのマッチ手法 | コメントごとに \s* を付与 | すべてまとめて \s* を後置 | 個々の空白処理が煩雑で見通しが悪い | より簡潔に複数の要素（--, /* */, (）を許容| 
| 3. 改行含むコメント対応 | .*?（ただし . は改行を含まない） | .*? + s フラグで改行含む | /* ... */ が複数行にまたがるとマッチしない | s フラグで . に改行含ませて完全対応| 
| 4. 単語境界の考慮 | なし（例：SELECTEDにもマッチする） | \b で明示 | SELECTED のようなtypoを誤判定の可能性あり | SELECT や WITH の 正確な単語 のみ判定| 
| 5. UNION やネスト対応 | 未対応（先頭が括弧だと失敗） | 対応（括弧スキップ） | (SELECT ...) UNION ...にマッチしない | スキップ対象に \( を追加して対応| 

どうでしょうか？